posthog/api/test/test_team.py

```diff
@@ -914,19 +915,88 @@ def test_can_set_and_unset_session_recording_network_payload_capture_config(self
 
     def test_can_set_and_unset_session_replay_config(self) -> None:
         # can set
-        first_patch_response = self.client.patch(
-            "/api/projects/@current/",
-            {"session_replay_config": {"record_canvas": True}},
+        self._patch_session_replay_config({"record_canvas": True})
+        self._assert_replay_config_is({"record_canvas": True})
+
+        # can unset
+        self._patch_session_replay_config(None)
+        self._assert_replay_config_is(None)
+
+    @parameterized.expand(
+        [
+            [
+                "string",
+                "Marple bridge",
+                "invalid_input",
+                "Must provide a dictionary or None.",
+            ],
+            ["numeric", "-1", "invalid_input", "Must provide a dictionary or None."],
+            [
+                "unexpected json - no record",
+                {"key": "something"},
+                "invalid_input",
+                "Must provide a dictionary with only allowed keys: included_event_properties, opt_in, preferred_events, excluded_events, important_user_properties.",
+            ],
+        ]
+    )
+    def test_invalid_session_replay_config_ai_config(
+        self, _name: str, provided_value: str, expected_code: str, expected_error: str
+    ) -> None:
+        response = self._patch_session_replay_config(
+            {"ai_config": provided_value}, expected_status=status.HTTP_400_BAD_REQUEST
         )
-        assert first_patch_response.status_code == status.HTTP_200_OK
+        assert response.json() == {
+            "attr": "session_replay_config",
+            "code": expected_code,
+            "detail": expected_error,
+            "type": "validation_error",
+        }
+
+    def test_can_set_and_unset_session_replay_config_ai_config(self) -> None:
+        # can set just the opt-in
+        self._patch_session_replay_config({"ai_config": {"opt_in": True}})
+        self._assert_replay_config_is({"ai_config": {"opt_in": True}})
+
+        # can set some preferences
+        self._patch_session_replay_config({"ai_config": {"opt_in": False, "included_event_properties": ["something"]}})
+        self._assert_replay_config_is({"ai_config": {"opt_in": False, "included_event_properties": ["something"]}})
+
+        self._patch_session_replay_config({"ai_config": None})
+        self._assert_replay_config_is({"ai_config": None})
+
+    def test_can_set_replay_configs_without_providing_them_all(self) -> None:
+        # can set just the opt-in
+        self._patch_session_replay_config({"ai_config": {"opt_in": True}})
+        self._assert_replay_config_is({"ai_config": {"opt_in": True}})
+
+        self._patch_session_replay_config({"record_canvas": True})
+        self._assert_replay_config_is({"record_canvas": True, "ai_config": {"opt_in": True}})
+
+    def test_can_set_replay_configs_must_send_all_nested_keys_though(self) -> None:
```

Comment:
The name of this test confused me for a little bit. I thought it was implying that you **had** to include every nested key in `ai_config` but I think what it's saying is that it overrides non-provided keys (if I understand it correctly).

Maybe a comment next to the second patch / assertion would be helpful